internal/credential/credential_libraries.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package credential
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+)
+
+// CredentialLibraryRepository defines the interface expected
+// to get the total number of credential libraries.
+type CredentialLibraryRepository interface {
+	GetTotalCredentialLibraries(context.Context) (int, error)
+	GetTotalSSHCertificateCredentialLibraries(context.Context) (int, error)
+	ListDeletedCredentialLibraryIds(context.Context, time.Time, ...Option) ([]string, error)
+	ListDeletedSSHCertificateCredentialLibraryIds(context.Context, time.Time, ...Option) ([]string, error)
```

Comment:
Note that the option types here must _not_ be the subtype repository option (e.g. Vault) as it leads to an import cycle. We could use a separate `internal/credentials/options` package, but this felt OK to me.